lib/mongo/cluster/topology/replica_set_with_primary.rb

```diff
@@ -0,0 +1,27 @@
+# Copyright (C) 2018 MongoDB, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+module Mongo
+  class Cluster
+    module Topology
+
+      # Defines behavior when a cluster is in replica set topology,
+      # and is a no primary which has been discovered by the driver.
+      #
+      # @since 2.7.0
+      class ReplicaSetWithPrimary < ReplicaSetNoPrimary
```

Comment:
Just to make sure I understand, are we planning on implementing any behavioral differences between ReplicaSetWithPrimary or ReplicaSetNoPrimary, or is the purpose of the separate classes just to make monitoring easier?

Reply:
The purpose right now is spec compliance which mandates the difference between the two. Depending on how much work/refactoring is done in the other tickets there may be behavior differences. For example, there shouldn't be a need to query servers for primary to find out if a topology has a primary.

Reply:
Makes sense